packages/analytics/shippers/elastic_v3/server/src/server_shipper.ts

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fetch from 'node-fetch';
+import {
+  BehaviorSubject,
+  filter,
+  Subject,
+  interval,
+  exhaustMap,
+  merge,
+  from,
+  firstValueFrom,
+  timer,
+  retryWhen,
+  tap,
+  delayWhen,
+  takeWhile,
+} from 'rxjs';
+import type {
+  AnalyticsClientInitContext,
+  Event,
+  EventContext,
+  IShipper,
+  TelemetryCounter,
+} from '@kbn/analytics-client';
+import { TelemetryCounterType } from '@kbn/analytics-client';
+import type { ElasticV3ShipperOptions } from '@kbn/analytics-shippers-elastic-v3-common';
+import {
+  buildUrl,
+  createTelemetryCounterHelper,
+  eventsToNDJSON,
+} from '@kbn/analytics-shippers-elastic-v3-common';
+
+const SECOND = 1000;
+const MINUTE = 60 * SECOND;
+const HOUR = 60 * MINUTE;
+const KIB = 1024;
+
+export class ElasticV3ServerShipper implements IShipper {
+  public static shipperName = 'elastic_v3_server';
+  public readonly telemetryCounter$ = new Subject<TelemetryCounter>();
+
+  private readonly reportTelemetryCounters = createTelemetryCounterHelper(
+    this.telemetryCounter$,
+    ElasticV3ServerShipper.shipperName
+  );
+
+  private readonly internalQueue: Event[] = [];
+  private readonly shutdown$ = new Subject<void>();
+
+  private readonly lastBatchSent$ = new BehaviorSubject<number>(Date.now());
```

Comment:
Looking at the usages in the file, I don't think you need an observable for this?